<?xml version="1.0"?>
<launch>
  <!-- GDB functionality -->
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />
  <arg     if="$(arg debug)" name="launch_prefix" value="gdb --ex run --args" />

  <arg name="frequency" default="30" />
  <arg name="pcan_device" default="/dev/pcanusb1" />

  <node name="schunk_canopen_node" pkg="schunk_canopen_driver" type="schunk_canopen_driver_node" output="screen" launch-prefix="$(arg launch_prefix)">
    <param name="frequency"           type="int"      value="$(arg frequency)" />
    <param name="can_device_name"          type="string"   value="$(arg pcan_device)"   />
    <rosparam command="load" file="$(find schunk_canopen_driver)/config/example_config.yaml"/>
    <param name="use_ros_control" type="bool" value="True"/>
  </node>

  <!-- robot model -->
  <param name="robot_description" command="$(find xacro)/xacro.py '$(find schunk_description)/urdf/lwa4p/arm.urdf.xacro'" />

  <!-- Load controller settings -->
  <rosparam file="$(find schunk_canopen_driver)/config/arm_controllers.yaml" command="load"/>

  <!-- spawn controller manager -->
  <node name="ros_control_controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
    output="screen" args="pos_based_pos_traj_controller" />

  <!-- load other controller -->
  <node name="ros_control_controller_manager" pkg="controller_manager" type="controller_manager" respawn="false"
    output="screen" args="load joint_state_controller" />

  <!--   Convert joint states to /tf tranforms -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>

</launch>
